core/string/ustring.h

```diff
@@ -623,6 +624,16 @@ _FORCE_INLINE_ String ETRN(const String &p_text, const String &p_text_plural, in
 	return p_text_plural;
 }
 
+// Internal function for configuration info macros.
+Dictionary _make_configuration_info(const String &p_message, const String &p_property_name, const String &p_severity);
+
+#define CONFIG_INFO(message) warnings.push_back(_make_configuration_info(message, "", "info"));
+#define CONFIG_INFO_P(message, property_name) warnings.push_back(_make_configuration_info(message, property_name, "info"));
+#define CONFIG_WARNING(message) warnings.push_back(_make_configuration_info(message, "", "warning"));
+#define CONFIG_WARNING_P(message, property_name) warnings.push_back(_make_configuration_info(message, property_name, "warning"));
+#define CONFIG_ERROR(message) warnings.push_back(_make_configuration_info(message, "", "error"));
+#define CONFIG_ERROR_P(message, property_name) warnings.push_back(_make_configuration_info(message, property_name, "error"));
```

Comment:
Looking back at my changes, not sure why I chose `ustring` for these defines.

I'd like to move these to a better suited place, but at the same time want to avoid updating the includes of every node in the repo. Suggestions?

Also while I'm at it, I should `#ifdef TOOLS_ENABLED`, since they shouldn't be used outside of a tools build anyway.

Reply:
Since I had to create a new `ConfigurationInfo` class on core (due to resources being core), it was a great place to move these macros into.